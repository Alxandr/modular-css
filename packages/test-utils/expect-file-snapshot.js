"use strict";

const fs = require("fs");

const { toMatchSnapshot } = require("jest-snapshot");

expect.extend({
    toMatchFileSnapshot(source, ...args) {
        return toMatchSnapshot.call(
            this,
            fs.readFileSync(source, "utf8"),
            ...args,
        );
    },
});
