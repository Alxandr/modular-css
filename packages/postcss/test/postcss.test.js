"use strict";

const fs = require("fs");
    
const postcss = require("postcss");
const namer = require("test-utils/namer.js");
const { cwd, find, temp, read } = require("test-utils/fixtures.js");

require("test-utils/expect-file-snapshot.js");

const plugin  = require("../postcss.js");

const process = (file, opts = {}) =>
    plugin.process(
        fs.readFileSync(file),
        Object.assign(
            Object.create(null),
            {
                from : file,
                cwd,
                namer,
            },
            opts
        )
    );

describe("/postcss.js", () => {
    it("should be a function", () => {
        expect(typeof plugin).toBe("function");
    });

    it("should process CSS and output the result", async () => {
        const result = await process(find("simple.css"));
        
        expect(result.css).toMatchSnapshot();
    });

    it("should process CSS with dependencies and output the result", async () => {
        const result = await process(find("start.css"));
        
        expect(result.css).toMatchSnapshot();
    });

    it("should process CSS and output exports as a message", async () => {
        const result = await process(find("simple.css"));
        
        expect(result.messages).toMatchSnapshot();
    });

    it("should accept normal processor options", async () => {
        const result = await process(find("simple.css"), {
            map : {
                inline : true,
            },
            namer : (f, s) => `fooga_${s}`,
        });

        expect(result.css).toMatchSnapshot();
    });

    it("should accept a `json` property and write exports to that file", async () => {
        await process(find("start.css"), {
            json : temp("classes.json"),
        });
        
        expect(temp("classes.json")).toMatchFileSnapshot();
    });

    it("should use output filepath for json if a custom path isn't provided", async () => {
        await process(find("start.css"), {
            json : true,
            to   : temp("start.css"),
        });
        
        expect(temp("start.json")).toMatchFileSnapshot();
    });

    it("should be usable like a normal postcss plugin", async () => {
        const processor = postcss([
            plugin({
                cwd,
                namer : () => "a",
            }),
        ]);
        
        const result = await processor.process(read("simple.css"), {
            from : find("simple.css"),
            map  : {
                inline : true,
            },
        });

        expect(result.css).toMatchSnapshot();
    });

    it("should output json when used within postcss", async () => {
        const processor = postcss([
            plugin({
                cwd,
                namer,
            }),
        ]);
        
        await processor.process(read("simple.css"), {
            from : find("simple.css"),
            json : temp("simple.json"),
        });

        expect(temp("simple.json")).toMatchFileSnapshot();
    });

    it("should accept json args in either position with postcss", async () => {
        var processor = postcss([
            plugin({
                cwd,
                namer,
                json : temp("simple.json"),
            }),
        ]);
        
        await processor.process(read("simple.css"), {
            from : find("simple.css"),
        });
        
        expect(temp("simple.json")).toMatchFileSnapshot();
    });
});
