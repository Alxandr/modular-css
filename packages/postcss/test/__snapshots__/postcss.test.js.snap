// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/postcss.js should accept a \`json\` property and write exports to that file 1`] = `
"{
    \\"fixtures/folder/folder.css\\": {
        \\"folder\\": \\"folder\\"
    },
    \\"fixtures/local.css\\": {
        \\"folder\\": \\"white\\",
        \\"one\\": \\"red\\",
        \\"two\\": \\"blue\\",
        \\"booga\\": \\"booga\\",
        \\"looga\\": \\"booga looga\\"
    },
    \\"fixtures/start.css\\": {
        \\"one\\": \\"red\\",
        \\"two\\": \\"blue\\",
        \\"folder\\": \\"white\\",
        \\"wooga\\": \\"booga wooga\\",
        \\"booga\\": \\"booga\\",
        \\"tooga\\": \\"tooga\\"
    }
}"
`;

exports[`/postcss.js should accept json args in either position with postcss 1`] = `
"{
    \\"fixtures/simple.css\\": {
        \\"str\\": \\"\\\\\\"string\\\\\\"\\",
        \\"fooga\\": \\"fooga\\"
    }
}"
`;

exports[`/postcss.js should accept normal processor options 1`] = `
".fooga_fooga {
    color: red;
}

/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImZpeHR1cmVzL3NpbXBsZS5jc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBRUE7SUFDSSxXQUFXO0NBQ2QiLCJmaWxlIjoiZml4dHVyZXMvc2ltcGxlLmNzcyIsInNvdXJjZXNDb250ZW50IjpbIkB2YWx1ZSBzdHI6IFwic3RyaW5nXCI7XG5cbi5mb29nYSB7XG4gICAgY29sb3I6IHJlZDtcbn1cbiJdfQ== */"
`;

exports[`/postcss.js should be usable like a normal postcss plugin 1`] = `
".a {
    color: red;
}

/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImZpeHR1cmVzL3NpbXBsZS5jc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBRUE7SUFDSSxXQUFXO0NBQ2QiLCJmaWxlIjoiZml4dHVyZXMvc2ltcGxlLmNzcyIsInNvdXJjZXNDb250ZW50IjpbIkB2YWx1ZSBzdHI6IFwic3RyaW5nXCI7XG5cbi5mb29nYSB7XG4gICAgY29sb3I6IHJlZDtcbn1cbiJdfQ== */"
`;

exports[`/postcss.js should output json when used within postcss 1`] = `
"{
    \\"fixtures/simple.css\\": {
        \\"str\\": \\"\\\\\\"string\\\\\\"\\",
        \\"fooga\\": \\"fooga\\"
    }
}"
`;

exports[`/postcss.js should process CSS and output exports as a message 1`] = `
Array [
  Object {
    "exports": Object {
      "fooga": Array [
        "fooga",
      ],
      "str": "\\"string\\"",
    },
    "type": "modular-css-exports",
  },
]
`;

exports[`/postcss.js should process CSS and output the result 1`] = `
".fooga {
    color: red;
}
"
`;

exports[`/postcss.js should process CSS with dependencies and output the result 1`] = `
".booga { color: red; background: blue; }
.tooga { border: 1px solid white; }
"
`;

exports[`/postcss.js should use output filepath for json if a custom path isn't provided 1`] = `
"{
    \\"fixtures/folder/folder.css\\": {
        \\"folder\\": \\"folder\\"
    },
    \\"fixtures/local.css\\": {
        \\"folder\\": \\"white\\",
        \\"one\\": \\"red\\",
        \\"two\\": \\"blue\\",
        \\"booga\\": \\"booga\\",
        \\"looga\\": \\"booga looga\\"
    },
    \\"fixtures/start.css\\": {
        \\"one\\": \\"red\\",
        \\"two\\": \\"blue\\",
        \\"folder\\": \\"white\\",
        \\"wooga\\": \\"booga wooga\\",
        \\"booga\\": \\"booga\\",
        \\"tooga\\": \\"tooga\\"
    }
}"
`;
